src/replica/python/schema_migration.py

```diff
@@ -80,6 +92,47 @@ def _set_version(self, version):
             raise RuntimeError(
                 f"Failed to update version number in database to {version}, current version is now {current}")
 
+    def _create_database(self):
+        """Create the replication controller database.
+        """
+        with closing(self.connection.cursor()) as cursor:
+            cursor.execute(f"CREATE DATABASE {self.database};")
+            warnings = cursor.fetchwarnings()
+            if warnings:
+                _log.warn("Warnings were creating database %s", self.database)
+        _log.info(f"Created database {self.database}.")
+        self.connection.commit()
+
+    def _create_users(self):
+        """Create the users for the replication controller database.
+        """
+        if not self.repl_connection:
+            raise RuntimeError(
+                "A non-admin replication database connection uri must be provided to initialize the "
+                "replication database."
+            )
+        user = make_url(self.repl_connection).username
+        if not user:
+            raise RuntimeError(
+                "To initialize the replication database, the non-admin connection uri must contain a user "
+                "name."
+            )
+        for stmt in [
+            f"CREATE USER IF NOT EXISTS {user}@localhost;",
+            f"CREATE USER IF NOT EXISTS {user}@'%';",
```

Comment:
Always creating accounts w/o passwords is not safe in some environments, especially because this account is granted all privileges for the Replication database. Here I'm more concerned about accidents (misconfigurations, etc) rather than malicious activities. Accidents of this kind may happen from time to time.

I think Fabrice always initializes database accounts with passwords, including this one.

Would it be possible to add the template placeholder for the password as well?

Reply:
Yes, but let's do it in a different ticket. I don't mind doing it next even, if you think it's high priority. But I want to defer, because this ticket, as-is unblocks Fabrice & will allow him to smig his existing databases to the latest version on Monday. This code, which is missing passwords, won't be run by him on Monday, because it is only run on a brand new, "Uninitialized" instance of qserv. So, it's not needed for what he will do.

Also, I'm not sure if template, or uri option is the correct source for the pw. Currently the {user} comes from the `--repl-connection` URI. Let's discuss it more Monday?

Reply:
I think it's easy to extract the password from the same connection string?
Though, if you need to defer this it's fine with me.